### PR TITLE
Add `PublicCode` of the journey to DatedJourney and ContinuousService

### PR DESCRIPTION
**Pull request by ue71603**
mostly as View to present things to customers

### CODE REVIEW

**Review by AndreasAtSBB — DISMISSED** [review]
PublishedJourneyName added after JourneyRef
operatorName added after OperatorRef

File: OJP/OJP_JourneySupport.xsd [hunk 1]
```diff
@@ -63,6 +63,11 @@
 		</xs:annotation>
 		<xs:sequence>
 			<xs:element ref="JourneyRef"/>
+			<xs:element name="PublishedJourneyName" type="xs:normalizedString" minOccurs="0">
```
Comment (Aurige) [inline OJP/OJP_JourneySupport.xsd:66]: What do you mean bt "published" ? why not just "JourneyName" ?

Comment (ue71603) [reply OJP/OJP_JourneySupport.xsd]: changed to PublicCode. Probably better as you suggested.

**Review by Aurige — DISMISSED** [review]

File: OJP/OJP_JourneySupport.xsd [hunk 2]
```diff
@@ -93,6 +98,11 @@
 				</xs:annotation>
 			</xs:element>
 			<xs:element ref="OperatorRef" minOccurs="0"/>
+			<xs:element name="OperatorName" type="xs:normalizedString" minOccurs="0">
```
Comment (sgrossberndt) [inline OJP/OJP_JourneySupport.xsd:101]: Should this be of type `InternationalTextStructure`?

File: OJP/OJP_JourneySupport.xsd [hunk 3]
```diff
@@ -63,6 +63,11 @@
 		</xs:annotation>
 		<xs:sequence>
 			<xs:element ref="JourneyRef"/>
+			<xs:element name="PublicCode" type="xs:string" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>Descriptive public code of the SERVICE JOURNEY.</xs:documentation>
```
Comment (sgrossberndt) [inline OJP/OJP_JourneySupport.xsd:68]: Could you please add an example?

File: OJP/OJP_JourneySupport.xsd [hunk 4]
```diff
@@ -63,6 +63,11 @@
 		</xs:annotation>
 		<xs:sequence>
 			<xs:element ref="JourneyRef"/>
+			<xs:element name="PublicCode" type="xs:string" minOccurs="0">
```
Comment (sgrossberndt) [inline OJP/OJP_JourneySupport.xsd:66]: Why is `OperatorName` below added as `type="xs:normalizedString` and `PublicCode` here as `type="xs:string"`? Wouldn't `type="xs:normalizedString` make sense here as well?

https://www.w3schools.com/xml/schema_dtypes_string.asp

File: OJP/OJP_JourneySupport.xsd (hunk 2 above)
Comment (ue71603) [reply OJP/OJP_JourneySupport.xsd]: updated

File: OJP/OJP_JourneySupport.xsd (hunk 3 above)
Comment (ue71603) [reply OJP/OJP_JourneySupport.xsd]: Not really. However, I harmonized with the NeTEx definition. Perhaps @Aurige has an example.

File: OJP/OJP_JourneySupport.xsd (hunk 4 above)
Comment (ue71603) [reply OJP/OJP_JourneySupport.xsd]: changed

**Review by Aurige — APPROVED** [review]

**Review by skinkie — APPROVED** [review]

**Review by sgrossberndt — CHANGES_REQUESTED** [review]
I still do not understand what a "public code" of a journey is. Is it some kind of ID? Should it be shown to customers? Is it meant to refer to journeys by it? If yes, in which context? Is this defined anywhere so we could add this information to the annotation?

**Review by Aurige — DISMISSED** [review]

**Comment by Aurige** [comment]
> I still do not understand what a "public code" of a journey is. Is it some kind of ID? Should it be shown to customers? Is it meant to refer to journeys by it? If yes, in which context? Is this defined anywhere so we could add this information to the annotation?

For example, in Ile de France, the urban trains (RER) have a short code, that is shown to the public (on timetable, displays on the platform and in vehicles) because there are a lot of different branches and destination and passengers need to be sure what train they are jumping in. This code is short enough to be shown even on old displays.

File: OJP/OJP_JourneySupport.xsd [hunk 5]
```diff
@@ -63,6 +63,11 @@
 		</xs:annotation>
 		<xs:sequence>
 			<xs:element ref="JourneyRef"/>
+			<xs:element name="PublicCode" type="xs:normalizedString" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>Public code of the SERVICE JOURNEY.</xs:documentation>
```
Comment (sgrossberndt) [inline OJP/OJP_JourneySupport.xsd:68]: As the element name "Public Code" has been chosen to be in line with NeTEx as @ue71603 wrote this should be reflected in the annotation like done in other elements, e.g.

```
<xs:element name="PublishedServiceName" type="InternationalTextStructure">
	<xs:annotation>
		<xs:documentation>Line name or service description as known to the public, f.e. "512", "S8" or "Circle Line" or "ICE 488".</xs:documentation>
	</xs:annotation>
</xs:element>
<xs:element name="TrainNumber" type="xs:normalizedString" minOccurs="0">
	<xs:annotation>
		<xs:documentation>Contains the TrainNumber description from NeTEx. If several TrainNumber types exist, use the commercial number.</xs:documentation>
	</xs:annotation>
</xs:element>
```

**Review by sgrossberndt — CHANGES_REQUESTED** [review]

File: OJP/OJP_JourneySupport.xsd (hunk 3 above)
Comment (ue71603) [reply OJP/OJP_JourneySupport.xsd]: changed

**Review by Aurige — APPROVED** [review]

**Comment by ue71603** [comment]
@sgrossberndt  Can you live with the clarification or still a problem?

**Review by sgrossberndt — APPROVED** [review]

**Pull request merged**